src/Client.php

```diff
@@ -125,6 +154,29 @@ private static function flatten(array $data, string $prefix = ''): array
         return $output;
     }
 
+    /**
+     * Retry a callback with exponential backoff
+     *
+     * @param callable $callback
+     * @return mixed
+     * @throws \Exception
+     */
+    private function retry(callable $callback): mixed
+    {
+        $attempts = 1;
+
+        while (true) {
+            $res = $callback();
+
+            if ($res->getStatusCode() !== 500 && $res->getStatusCode() !== 503 || $attempts >= $this->maxRetries) {
```

Comment:
Let's make the codes configurable (`setRetryStatusCodes`), some clients may want to retry on different codes